src/sessions/base-session-handler.ts

```diff
@@ -222,14 +255,13 @@ export default abstract class BaseSessionHandler {
     await Promise.all(destroyMediaPromises);
 
     const newStartMediaContraints = {
-      audio: options.audioDeviceId,
-      /* if video is muted, we don't want to request it */
-      video: !session.videoMuted && options.videoDeviceId,
+      audio: updateAudio && options.audioDeviceId,
+      video: updateVideo && options.videoDeviceId,
       session
     };
 
     /* Allow null because it is the system default. If false or undefined, do not request media. */
-    if (!stream && (newStartMediaContraints.audio || newStartMediaContraints.video) || (newStartMediaContraints.audio === null || newStartMediaContraints.video === null)) {
+    if (!stream && (updateAudio || updateVideo)) {
       try {
         stream = await this.sdk.media.startMedia(newStartMediaContraints);
```

Comment:
Do we want to skip this if we aren't updating audio or video? Otherwise it will throw an error down the line right?

Reply:
We are on line 264 `... && (updateAudio || updateVideo)`. One of those has to be true if we ever want to request media.